# This notebook corresponds to VAIL Day 15 Activity

In [1]:
#making imports

import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
from sklearn.linear_model import LogisticRegression
from sklearn.model_selection import train_test_split

In [2]:
#creating a voices data_frame

voices_df = pd.read_csv('voice.csv')

#showing the head of the data frame
print(voices_df.head())

   meanfreq        sd    median       Q25       Q75       IQR       skew  \
0  0.059781  0.064241  0.032027  0.015071  0.090193  0.075122  12.863462   
1  0.066009  0.067310  0.040229  0.019414  0.092666  0.073252  22.423285   
2  0.077316  0.083829  0.036718  0.008701  0.131908  0.123207  30.757155   
3  0.151228  0.072111  0.158011  0.096582  0.207955  0.111374   1.232831   
4  0.135120  0.079146  0.124656  0.078720  0.206045  0.127325   1.101174   

          kurt    sp.ent       sfm  ...  centroid   meanfun    minfun  \
0   274.402906  0.893369  0.491918  ...  0.059781  0.084279  0.015702   
1   634.613855  0.892193  0.513724  ...  0.066009  0.107937  0.015826   
2  1024.927705  0.846389  0.478905  ...  0.077316  0.098706  0.015656   
3     4.177296  0.963322  0.727232  ...  0.151228  0.088965  0.017798   
4     4.333713  0.971955  0.783568  ...  0.135120  0.106398  0.016931   

     maxfun   meandom    mindom    maxdom   dfrange   modindx  label  
0  0.275862  0.007812  0.007812  

In [3]:
#parsing x/y data so we can test,train,split data

X_data, Y_data =  voices_df.iloc[:, 0:20], voices_df['label']

X_train, X_test, y_train, y_test = train_test_split(X_data, Y_data, test_size=0.3, shuffle=True)

In [4]:
#importing logistic regression model from skl

model = LogisticRegression(solver='lbfgs',max_iter=1000)
model.fit(X_train,y_train)

LogisticRegression(max_iter=1000)

In [5]:
model.score(X_test,y_test)

0.9064143007360673

Scoring Notes:
  * max_iters=1000, solver='lbfgs' --> gender classification accuracy of 0.91
  * max_iters=1000, solver='newton-cg' --> gender classification accuracy of 0.91
  * max_iters=1000, solver='liblinear' --> agender classification ccuracy of 0.91
